step/serialization.md

```diff
@@ -2,36 +2,43 @@
 
 ## Overview
 
-Serialization has long been a sticky subject in Java. Brian Goetz describing it as a paradox, a key to Java's success allowing for transparent remoting, but also the implementation of serialization making a number of key mistakes. 
+Serialization has long been a sticky subject in Java. Brian Goetz describing it as a paradox, a key to Java's success allowing for transparent remoting, but also the implementation of serialization having a number of critical issues. 
 
-A short list of mistakes include;
-* Bypasses constructors and encapsulation  
+A short list of issues include;
+
+* Breaks encapsulation by ignoring constructors and supplying values to fields via reflection 
 * The behavior of serialization depends on "magic" methods (including `readObject()` which we will look at later)
 * Poor stream format that is not efficent, reusable, nor human readable 
 
-In this section we will explore some of the short comings with Serialization in Java, how some of these issues have been addressed in recent JEPs, and how Records addresses many of the issues with serialization. 
+In this lab we will explore in-practice some of the short comings with serialization in Java, how some of these issues have been addressed in recent JEPs, and how Records addresses many of the remaining issues with serialization. 
+
+## What is Serialization and Deserialization?
+
+Serialization is the process of transforming an object into a bytestream. Appropriately the process of deserialization is the inverse, where a bytestream is converted into a Java object.
+
+As mentioned in the intro, the implemention of serialization as a core Java feature was vital to its raise in popularity. It allowed for the easy transmission and receiving of data to a remote processes, persisting state to disk, and communication with external services like a database. Historically this required writing a lot of complex code that was difficult to write and read, and was often error-prone. 
```

Comment:
remove "As mentioned in the intro"

Reply:
I wouldn't use the past tense as serilziation is still used and useful today.

Reply:
"Historically this required writing a lot of complex code", please rephrase as using serilziation is easy, the complex part is to write **secure** code that use serilziation